timbermill-java/timbermill-server/src/main/java/com/datorama/timbermill/server/service/RequestWrapperFilter.java

```diff
@@ -11,11 +11,26 @@
 import org.springframework.web.filter.OncePerRequestFilter;
 import org.springframework.web.util.ContentCachingRequestWrapper;
 
+import com.google.common.net.HttpHeaders;
+
+import kamon.metric.Timer;
+import static com.datorama.oss.timbermill.common.KamonConstants.GZIP_DECOMPRESS_REQUEST_DURATION;
+
 @Component
 public class RequestWrapperFilter extends OncePerRequestFilter {
 
 	@Override
 	protected void doFilterInternal(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, FilterChain filterChain) throws ServletException, IOException {
-		filterChain.doFilter(new ContentCachingRequestWrapper(httpServletRequest), httpServletResponse);
+
```

Comment:
decide according to header if it has gzip payload or not , and wrap with appropriate wrapper accordingly